docs/languages/powershell.md

```diff
@@ -0,0 +1,207 @@
+
+Windows 10 and Server 2016 comes with Pester 3.40 pre-installed.
+to update Pester or to install the latest version on other platforms follow the [Pester installation instructions](https://github.com/pester/Pester#installation)
+
+![Pester CodeLens Integration](./images/powershell/pesterCodeLens.png)
+
+## Plaster
+
+ Plaster is a template-based file and project generator written in PowerShell. Its purpose is to streamline the creation of PowerShell module projects, Pester tests, DSC configurations and more.
+ See the [Github Plaster repo](https://github.com/PowerShell/Plaster) for more information.
+
+ [Cmdlet Documentation](https://github.com/PowerShell/Plaster/blob/master/docs/en-US/Plaster.md)
+
+ Create a new Plaster Project from the [command pallette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette)
+( opened with <kbd>ctrl</kbd>-<kbd>shift</kbd>-<kbd>p</kbd>, or <kbd>cmd</kbd>-<kbd>shift</kbd>-<kbd>p</kbd> on macOS) and typing Plaster:
+![Plaster Project](./images/powershell/cpPlasterCommand.png)
+
+
+## PSScriptAnalyzer
+
+The PowerShell Extension includes PSSCriptAnalyzer by default, and automatically performs analysis on PowerShell script files being edited in VSCode.
+
+[PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer#introduction) is a static code checker for PowerShell modules and scripts. PSScriptAnalyzer checks the quality of PowerShell code by running a set of rules. The rules are based on PowerShell best practices identified by the PowerShell Team and the community. Pester generates diagnostic rescords (errors and warnings) to inform users about potential code defects and suggests possible solutions for improvements.
+
+### PSScriptAnalyzer Settings
+
+PSScriptAnalyzer is shipped with a collection of built-in rules that checks various aspects of PowerShell code such as presence of uninitialized variables, usage of PSCredential Type, usage of Invoke-Expression etc. additional functionalities such as exclude/include specific rules are also supported.
+
+To disable PSSCriptAnalyzer open the settings, browse extensions, select the PowerShell Extension and deselect the checkbox for
+
+![PSScriptAnalyzer Settings](./images/powershell/pssaExtensionSetting.png)
+
+Format document command is provided by the PSScriptAnalyzer module.
+
+### Document Formatting
+
+Automatic document formatting can be invoked from the [command pallette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette)
+by entering Format keyword or using the <kbd>alt</kbd>+<kbd>shift></kbd>+<kbd>F</kbd>  keyboard shortcut.
+
+## CodeLens
+
+Shows actionable, contextual information that is displayed within the source code.
```

Comment:
To make this a full sentence I would start it with "Code Lenses are a VSCode feature to provide contextual, actionable information..."